### PR TITLE
 Upgrade Drupal Core to 10.

### PR DESCRIPTION
**Pull request by vishalkhode1**
**Motivation**
Fixes: [ACMS-1629](https://backlog.acquia.com/browse/ACMS-1629).

**Proposed changes**

- Removed upgrade_status module.
- Minimum required `PHP` is PHP 8.1.
- Minimum required `drupal/core-composer-scaffold` & `drupal/core-recommended ` is 10.0.
- Drupal core upgraded to 10.0.1.
- Dropped 10.x support for `drush/drush`.
- Dropped 1.x support for `oomphinc/composer-installers-extender`.

**Alternatives considered**
N/A.

**Testing steps**
- CI tests should pass.
- Manual Testing.

### CODE REVIEW

**Comment by danepowell** [comment]
@secretsayan this has a high probability of causing an ORCA outage, please review and coordinate with @vishalkhode1 to ensure stability. At a minimum, ORCA will probably need to modify the PHP version constraints in composer.json when building a fixture.

Apart from that, looks good from my side.

**Comment by danepowell** [comment]
The `require_label` test should have re-run and passed when I applied the enhancement label... not sure why that didn't happen, maybe GitHub Actions is slow today.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -4,3 +4,13 @@ updates:
     directory: "/"
     schedule:
       interval: "daily"
+  - package-ecosystem: "composer"
```
Comment (vishalkhode1) [inline .github/dependabot.yml:7]: This is done, so the dependabot doesn't try to upgrade Drupal core to 10 for 1.x branch.
cc: @danepowell / @secretsayan

**Review by secretsayan — APPROVED** [review]

**Comment by danepowell** [comment]
It looks like @secretsayan removed http-hmac-php, not me: https://github.com/acquia/drupal-recommended-project/commit/267d70747010acb84d8dd0df38d2722edaec99d9

Anyway if you look at the PR where this change happened, I explicitly comment on why it's removed: https://github.com/acquia/drupal-recommended-project/pull/251#issuecomment-1320091770

tl;dr it's not needed as a direct dependency.

**Comment by vishalkhode1** [comment]
> It looks like @secretsayan removed http-hmac-php, not me: [267d707](https://github.com/acquia/drupal-recommended-project/commit/267d70747010acb84d8dd0df38d2722edaec99d9)
> 
> Anyway if you look at the PR where this change happened, I explicitly comment on why it's removed: [#251 (comment)](https://github.com/acquia/drupal-recommended-project/pull/251#issuecomment-1320091770)
> 
> tl;dr it's not needed as a direct dependency.

@danepowell Thanks for the confirmation. I'm removing it from here as well then.
cc: @secretsayan
